### DISCUSSION TITLE
[Q] How to disable global app keys on screen?

### DISCUSSION THREAD

**Discussion by alexveden** [Textualize/textual#1745] [Q&A] [answered]
Is it possible to disable app key bindings, and keep only specific screen bindings when it's active?

In my case, I have an app, which uses hot keys to navigate across screens (a.k.a. text tabs). Each screen has dedicated bindings for operations and <Esc> to close it. However, after screen pop up the app is handling global bindings, but I'd like to allow only use <Esc> to back to the main screen and disallow global bindings.

Footer widget also gets cluttered for a child screens because also contains app bindings.

Any suggestions?

![image](https://user-images.githubusercontent.com/18488560/217743280-2bf5caf1-7b4c-4a2a-91cb-661415735cbf.png)

**Comment by davep** [comment, marked as answer]
Likely your best approach here is to not bind those keys on the app, and to not use the default screen as your main screen. Instead supply your own main screen and bind the keys on that. 

See https://textual.textualize.io/blog/2022/11/26/on-dog-food-the-original-metaverse-and-not-being-bored/#dont-use-the-default-screen for example.

**Comment by alexveden** [comment]
@davep thanks, actually having an explicit main screen makes much more sense and clarity in the code. Possibly this should be crystallized as a best practice recommendation in documentation down the road.

**Reply by davep** [reply]
Aye, once things start to settle down and we get a proper appreciation of how people build with Textual I can see this happening. We do aim, at some point, to have a slightly opinionated "here's how to get your app going" tutorial for non-trivial apps.
